steer_drive_controller/include/steer_drive_controller/steer_drive_controller.h

```diff
@@ -0,0 +1,226 @@
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the PAL Robotics nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/*
+ * Author: Enrique Fernández
+ * Author: Masaru Morita
+ */
+
+#include <controller_interface/controller.h>
+#include <controller_interface/multi_interface_controller.h>
+#include <hardware_interface/joint_command_interface.h>
+#include <pluginlib/class_list_macros.h>
+
+#include <nav_msgs/Odometry.h>
+#include <tf/tfMessage.h>
+
+#include <realtime_tools/realtime_buffer.h>
+#include <realtime_tools/realtime_publisher.h>
+
+#include <steer_drive_controller/odometry.h>
+#include <steer_drive_controller/speed_limiter.h>
+
+//#define GUI_DEBUG // uncommentout when you use qtcreator for debugging
+
+namespace steer_drive_controller{
+
+  /**
+   * This class makes some assumptions on the model of the robot:
+   *  - the rotation axes of wheels are collinear
+   *  - the wheels are identical in radius
+   * Additional assumptions to not duplicate information readily available in the URDF:
+   *  - the wheels have the same parent frame
+   *  - a wheel collision geometry is a cylinder in the urdf
+   *  - a wheel joint frame center's vertical projection on the floor must lie within the contact patch
+   */
+  class SteerDriveController
+      : public controller_interface::MultiInterfaceController<
+      hardware_interface::PositionJointInterface,
+      hardware_interface::VelocityJointInterface>
+  {
+    // constants
+  private:
+    enum INDX_WHEEL {
```

Comment:
I couldn't find a reference to these indices anywhere. It's also bad practice to rely on positions in arrays.

Reply:
Ok, I'll remove INDX_WHEEL indices. Could you please which way is better to specify positions without arrays? I would like to follow a better way if possible.

Reply:
The general practice where you'd use arrays is when you have something global or semi-global storing all the data and then parts of the code "magically know" which index they need. A practice that limits the scope of functions is passing [handles](https://stackoverflow.com/questions/13023405/what-is-the-difference-between-handle-pointer-and-reference) to data a given function needs to have access to. I'm not sure this directly applies here, as in this case the issue I raised was that there's simply no reference to those enum indices :)

Reply:
Fixed in https://github.com/MoriKen254/ros_controllers/commit/4abe230852452bda33e169c76fbe4567ba27c6d9.